db/primary_migrate/20240321130431_add_sign_in_new_device_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddSignInNewDeviceToUsers < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :sign_in_new_device_at, :datetime
```

Comment:
I think we'll want to have an index on this column if we're performing queries across the entire users table based on the value, similar to how `account_reset_requests` has indices on its date fields to facilitate similar querying in `AccountReset::GrantRequestsAndSendEmails`.